packages/block-editor/src/components/editor-styles/index.js

```diff
@@ -1,62 +1,52 @@
 /**
  * External dependencies
  */
-import { compact, map } from 'lodash';
 import tinycolor from 'tinycolor2';
 
 /**
  * WordPress dependencies
  */
-import { useCallback, useRef } from '@wordpress/element';
+import { useCallback } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
 import transformStyles from '../../utils/transform-styles';
 
-function syncDarkThemeBodyClassname( node ) {
-	const backgroundColor = window
-		.getComputedStyle( node, null )
-		.getPropertyValue( 'background-color' );
-
-	const { ownerDocument } = node;
-	const body = ownerDocument.getElementsByTagName( 'body' )[ 0 ];
-
-	if ( tinycolor( backgroundColor ).getLuminance() > 0.5 ) {
-		body.classList.remove( 'is-dark-theme' );
-	} else {
-		body.classList.add( 'is-dark-theme' );
-	}
+const EDITOR_STYLES_CLASS_NAME = '.editor-styles-wrapper';
+
+function useDarkThemeBodyClassName( dependencies ) {
+	return useCallback( ( node ) => {
+		if ( ! node ) {
+			return;
+		}
+
+		const { ownerDocument } = node;
+		const { defaultView, body } = ownerDocument;
+		const canvas = ownerDocument.querySelector( EDITOR_STYLES_CLASS_NAME );
+		const backgroundColor = defaultView
+			.getComputedStyle( canvas, null )
+			.getPropertyValue( 'background-color' );
+
+		if ( tinycolor( backgroundColor ).getLuminance() > 0.5 ) {
+			body.classList.remove( 'is-dark-theme' );
+		} else {
+			body.classList.add( 'is-dark-theme' );
+		}
+	}, dependencies );
 }
 
-export default function useEditorStyles( styles ) {
-	const nodes = useRef( [] );
-
-	return useCallback(
-		( node ) => {
-			if ( ! node ) {
-				nodes.current.forEach( ( styleElement ) =>
-					styleElement.ownerDocument.body.removeChild( styleElement )
-				);
-				return;
-			}
-
-			const updatedStyles = transformStyles(
-				styles,
-				'.editor-styles-wrapper'
-			);
-
-			const { ownerDocument } = node;
-			nodes.current = map( compact( updatedStyles ), ( updatedCSS ) => {
-				const styleElement = ownerDocument.createElement( 'style' );
-				styleElement.innerHTML = updatedCSS;
-				ownerDocument.body.appendChild( styleElement );
-
-				return styleElement;
-			} );
-
-			syncDarkThemeBodyClassname( node );
-		},
-		[ styles ]
+export default function EditorStyles( { styles } ) {
+	return (
+		<>
+			{ /* Use an empty style element to have a document reference,
+			     but this could be any element. */ }
+			<style ref={ useDarkThemeBodyClassName( styles ) } />
```

Comment:
I think this is wrong? Should be `[ styles ]`. Or am I missing something?

In practice, passing dependencies variable around is rarely a good practice, as it's very easy to miss these kinds of bugs. That's also why #24914 disable this usage. Writing them all explicitly is not a bad idea IMO.

`useDarkThemeBodyClassName` could probably just be written as:

```js
function useDarkThemeBodyClassName(styles) {
  return useCallback(
    // ...
  , [styles]);
}
```

Reply:
`styles` is an array so it would only update if one of the values changes. Not sure when the array reference itself changes. I'll change it back to what it was.